### DISCUSSION TITLE
Langauge-server not starting with proper schema file

### DISCUSSION THREAD

**Discussion by addisonj** [markdoc/markdoc#500] [Bug report]
I am working with the markdoc language server but am unable to get it to start with the following exception:

```
TypeError: Cannot read properties of undefined (reading 'markdoc')
    at new Documents (/Users/addisonj/.vscode/extensions/stripe.markdoc-language-support-0.0.12/dist/client/server.js:27176:52)
    at /Users/addisonj/.vscode/extensions/stripe.markdoc-language-support-0.0.12/dist/client/server.js:27434:7
    at Array.map (<anonymous>)
    at server (/Users/addisonj/.vscode/extensions/stripe.markdoc-language-support-0.0.12/dist/client/server.js:27432:41)
```

Looking at previous issues, it seems like in most cases, this is due to an invalid schema file that cannot be loaded into a vanilla node process. Here is the relevant context:

markdoc.config.js:

```json
[
  {
    "id": "knowledge-hub",
    "path": "src/docs",
    "schema": {
      "path": "dist/markdoc/lang-server.js",
      "type": "esm",
      "property": "default",
      "watch": true
    }
  }
]
```

My lang-server.js file:

```js
import tags from './tags/index.js';
import nodes from './nodes/index.js';
const config = {
    tags,
    nodes,
};
export default config;
```

And here I can see that being imported into node:

```bash
$ node
Welcome to Node.js v20.11.0.
Type ".help" for more information.
> const { default: config} = await import('./dist/markdoc/lang-server.js')
undefined
> config
{
  tags: {
    callout: { attributes: [Object], render: 'Callout' },
    comment: { render: undefined },
    break: { render: 'LineBreak' },
    card: { render: 'DocCard', attributes: [Object] },
    'card-grid': { children: [Array], render: 'DocCardGroup' },
    carousel: { children: [Array], render: 'DocCarousel', attributes: {} },
    'carousel-item': { render: 'DocCarouselItem', attributes: {} },
    hovertip: {
      render: 'HoverTip',
      attributes: [Object],
      validate: [Function: validate],
      transform: [Function: transform]
    },
    tab: { render: 'TabItem', attributes: [Object] },
    tabs: {
      children: [Array],
      render: 'Tabs',
      attributes: {},
      transform: [Function: transform]
    },
    side: { render: 'Side' },
    'side-by-side': { children: [Array], render: 'SideBySide', attributes: {} },
    youtube: { render: 'YouTube', attributes: [Object] }
  },
  nodes: {
    document: { render: undefined, attributes: [Object] },
    comment: { render: undefined },
    heading: {
      children: [Array],
      attributes: [Object],
      transform: [Function: transform]
    },
    th: { render: 'th', attributes: [Object] }
  }
}
```

I have attempted to trace the code and see if I get any exceptions in loading the file, but so far, I haven't been able to even see the [load function](https://github.com/markdoc/language-server/blob/a4ba348b889f51e6a72c8fe8642796a95e1d56c7/server/services/schema.ts#L43) even being called! I did this by manually modifying my locally installed copy of the extension and just added a `console.log` to the top of that load function.

I have also done the same trick for seeing where the exception is thrown and added logging to [the server function](https://github.com/markdoc/language-server/blob/a4ba348b889f51e6a72c8fe8642796a95e1d56c7/server/server.ts#L48) and the object I get back from the `connect` function does *not* have a config key and instead the options object *only* has a `dispose` function.

I actually tried this with a very basic project and had the same issues, so I am wondering if there is some something higher up in the chain that is causing these issues!

**Comment by addisonj** [comment]
Thinking about it a bit more, I am realizing I probably am never loading my schema file at all because I am not even able to load the config.

Looking at recent changes in the `0.0.12` release, I see this: https://github.com/markdoc/language-server/commit/3f53a1766db808fca3318f87dcf4fe3238b24192#diff-cfbaa1474fd5b35981f53d787fd188be4fe43c9d9f68525c67ff6e0fe2934aad change, which appears to now override the `options` object to be the result of the `connection.onInitialized()` object, which, looking at the type from the [vscode lsp server](https://github.com/microsoft/vscode-languageserver-node/blob/6051b53c343d8736c468bd6e41a7ad2ed8bc1f09/server/src/common/server.ts#L1016) is a `Disposable`

So I am fairly certain the linked recent change is a bug. I will attempt to rollback (how to do that... 🤔) and see if that fixes the issue and report back

**Comment by addisonj** [comment]
Okay, I confirmed this is just a bug due to recent regression in the language server code, I have created the following pull request: 

https://github.com/markdoc/language-server/pull/19

**Reply by louiss0** [reply]
> Okay, I confirmed this is just a bug due to recent regression in the language server code, I have created the following pull request:
> 
> [markdoc/language-server#19](https://github.com/markdoc/language-server/pull/19)

My markdoc code is not working still. I don't know why but I also got similar errors. 
I tried to install past versions but only `0.1.2` showed me an error.

**Reply by mfix-stripe** [reply]
PTAL  @segphault
